### ISSUE TITLE
obs interconnect doesn't honor limit parameter

### ISSUE THREAD

**Issue by lnussel** [openSUSE/open-build-service#2545]
$ osc api '/source/openSUSE:Factory/postfix/_history?limit=5'|grep -c vrev
5
$ osci api '/source/openSUSE.org:openSUSE:Factory/postfix/_history?limit=5'|grep -c vrev
135

**Comment by lnussel** [comment]
affects SLE review bots

**Comment by ChrisBr** [comment]
Looked into it and seems like a backend issue, in both cases we pass the limit=5 parameter to the backend. @M0ses time to have a look?

**Comment by M0ses** [comment]
@ChrisBr 

The request hitting the source server does not contain the limit parameter in case of /public route (which is used by interconnect).

SEE
`curl https://api.opensuse.org/public/source/openSUSE:Factory/postfix/_history?limit=5`

So my wild guess would be, that the limit parameter is not forwarded to the source server by the api.

I tested this monkey patch, and it worked for me:

```
diff --git a/src/api/app/controllers/public_controller.rb b/src/api/app/controllers/public_controller.rb
index ccecdc2..e734860 100644
--- a/src/api/app/controllers/public_controller.rb
+++ b/src/api/app/controllers/public_controller.rb
@@ -106,7 +106,7 @@ class PublicController < ApplicationController
 
     path = "/source/#{CGI.escape(params[:project])}/#{CGI.escape(params[:package])}/#{CGI.escape(file)}"
 
-    path += build_query_from_hash(params, [:rev])
+    path += build_query_from_hash(params, [:rev, :limit])
     volley_backend_path(path) unless forward_from_backend(path)
   end
```

**Comment by adrianschroeter** [comment]
Marco is working on this already

**Comment by ChrisBr** [comment]
@M0ses ah thanks! I didn't look into the public, makes sense!

**Comment by lethliel** [comment]
solved with #2674
